test/unittests/api/v0.0.1/candidates/Visit_Test.php

```diff
@@ -89,6 +89,7 @@ function testPutInvalidBattery() {
         $this->assertEquals($API->Headers, ["HTTP/1.1 400 Bad Request"]);
     }
     function testPostValidVisit() {
+        \User::singleton("admin");
```

Comment:
The rest of the unit test will break if this is not done. Like I said, badly structured code/unit tests.

We don't have code anywhere that automatically logs a user in for unit testing. We don't seem to set `User::singleton()` to anything for unit testing. Meaning, we just simply assume there is no logged in user. This unit test requires a user to be logged in.